datasets/meta_woz/meta_woz.py

```diff
@@ -0,0 +1,153 @@
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "dialogues"
+
+    def _info(self):
+        if self.config.name == "tasks":
+            features = datasets.Features(
+                {
+                    "task_id": datasets.Value("string"),
+                    "domain": datasets.Value("string"),
+                    "bot_prompt": datasets.Value("string"),
+                    "bot_role": datasets.Value("string"),
+                    "user_prompt": datasets.Value("string"),
+                    "user_role": datasets.Value("string"),
+                }
+            )
+        else:
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "user_id": datasets.Value("string"),
+                    "bot_id": datasets.Value("string"),
+                    "domain": datasets.Value("string"),
+                    "task_id": datasets.Value("string"),
+                    "turns": datasets.Sequence(datasets.Value("string")),
+                }
+            )
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download(_URLs)
+        data_dir = dl_manager.extract(data_dir)
+        data_dir["test"] = dl_manager.extract(os.path.join(data_dir["test"], "dstc8_metalwoz_heldout.zip"))
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"data_dir": data_dir["train"]},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"data_dir": data_dir["test"]},
+            ),
+        ]
+
+    def _generate_examples(self, data_dir):
+        """ Yields examples. """
+        if self.config.name == "tasks":
+            filepath = os.path.join(data_dir, "tasks.txt")
+            with open(filepath, encoding="utf-8") as f:
+                for id_, row in enumerate(f):
+                    data = json.loads(row)
+                    yield id_, {
+                        "task_id": data["task_id"],
+                        "domain": data["domain"],
+                        "bot_prompt": data["bot_prompt"],
+                        "bot_role": data["bot_role"],
+                        "user_prompt": data["user_prompt"],
+                        "user_role": data["user_role"],
+                    }
+        else:
+            id_ = -1
+            base_path = os.path.join(data_dir, "dialogues")
+            file_list = [os.path.join(base_path, file) for file in os.listdir(base_path)]
```

Comment:
```suggestion
            file_list = sorted([os.path.join(base_path, file) for file in os.listdir(base_path)])
```
we need to make the order deterministic here